samza-core/src/main/java/org/apache/samza/operators/spec/OperatorSpec.java

```diff
@@ -61,9 +62,15 @@
   /**
    * The set of operators that consume the messages produced from this operator.
    * <p>
-   * We use a LinkedHashSet since we need deterministic ordering in initializing/closing operators.
+   * We use a LinkedHashSet since we need both deterministic ordering in initializing/closing operators and serializability.
    */
-  private final Set<OperatorSpec<OM, ?>> nextOperatorSpecs = new LinkedHashSet<>();
+  private final LinkedHashSet<OperatorSpec<OM, ?>> nextOperatorSpecs = new LinkedHashSet<>();
+
+  @VisibleForTesting
+  final boolean isClone(OperatorSpec other) {
```

Comment:
Is this still required?

Reply:
As annotation indicates, this method is only for unit test usage now.

Reply:
Ah ok. Can we add a comment that this is only for testing instead of the annotation? VisibleForTesting usually indicates that the method has a looser access restriction (e.g. protected or public instead of private) for testing purposes, but should otherwise be considered private.